x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceNamedWriteablesProvider.java

```diff
@@ -198,34 +198,34 @@ private static void addMistralNamedWriteables(List<NamedWriteableRegistry.Entry>
         // note - no task settings for Mistral embeddings...
     }
 
-    private static void addAzureAiStudioNamedWriteables(List<NamedWriteableRegistry.Entry> namedWriteables) {
+    private static void addAzureAiFoundryNamedWriteables(List<NamedWriteableRegistry.Entry> namedWriteables) {
         namedWriteables.add(
             new NamedWriteableRegistry.Entry(
                 ServiceSettings.class,
-                AzureAiStudioEmbeddingsServiceSettings.NAME,
-                AzureAiStudioEmbeddingsServiceSettings::new
+                AzureAiFoundryEmbeddingsServiceSettings.NAME,
+                AzureAiFoundryEmbeddingsServiceSettings::new
             )
         );
         namedWriteables.add(
             new NamedWriteableRegistry.Entry(
                 TaskSettings.class,
-                AzureAiStudioEmbeddingsTaskSettings.NAME,
-                AzureAiStudioEmbeddingsTaskSettings::new
+                AzureAiFoundryEmbeddingsTaskSettings.NAME,
+                AzureAiFoundryEmbeddingsTaskSettings::new
             )
         );
 
         namedWriteables.add(
             new NamedWriteableRegistry.Entry(
                 ServiceSettings.class,
-                AzureAiStudioChatCompletionServiceSettings.NAME,
-                AzureAiStudioChatCompletionServiceSettings::new
+                AzureAiFoundryChatCompletionServiceSettings.NAME,
```

Comment:
NAME has changed from `azure_ai_studio_chat_completion_service_settings` to `azure_ai_foundry_chat_completion_service_settings`. When an old node writes a named writable with the old name `azure_ai_studio_chat_completion_service_settings` this node will not know about it. 

Because there is logic in the `AzureAiFoundryChatCompletionServiceSettings` serialisation code to handle backwards compatibility it is better not to change the name. Just add a comment explaining that `NAME` hasn't changed to maintain BWC